pkg/prometheus/operator.go

```diff
@@ -146,6 +146,7 @@ func ValidateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
 	return nil
 }
 
+// TODO: Add validation for the relabel configs
```

Comment:
I would perform the validation here
https://github.com/prometheus-operator/prometheus-operator/blob/ffaeca223cb8a5626e71b970910aa07a878c9161/pkg/prometheus/server/operator.go#L1127-L1131

BTW there's no hard reason why `AddAlertmanagerEndpointsToStore()`  (and nested functions) lives in pkg/prometheus since it's only called from pkg/prometheus/server.

Reply:
Would you want me to add it to `ValidateAlertmanagerEndpoints` (which is a nested-function)? Or do you mean to add a separate function for validating relabel configs?

Reply:
I have added it as part of `ValidateAlertmanagerEndpoints()`.

Reply:
I removed the validation from the `AddTo...Store` function and moved it in the if block `if Spec.Alerting !=nil`, alongside `AddTo...Store`.